File: BackEnd.Net/src/Alcohol/Models/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Alcohol.Models
{
    public class Category
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [StringLength(500)]
        public string Description { get; set; }

        [Required]
        [StringLength(100)]
        public string Slug { get; set; }

        [Required]
        public int DisplayOrder { get; set; }
        public string ImageUrl { get; set; }
        
        [Required]
        public bool IsActive { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public int? ParentId { get; set; }

        // Navigation properties
        [ForeignKey("ParentId")]
        public Category Parent { get; set; }
        public ICollection<Category> Children { get; set; }
        public ICollection<Product> Products { get; set; }

        public Category()
        {
            Children = new HashSet<Category>();
            Products = new HashSet<Product>();
        }
    }
} 

File: FrontEndNext/frontendnext/src/lib/types/category.ts
export interface Category {
  id: number;
  name: string;
  description: string;
  slug: string;
  parentId: number | null;
  parentName: string | null;
  parentSlug: string | null;
  displayOrder: number;
  isActive: boolean;
  imageUrl: string | null;
  metaTitle: string | null;
  metaDescription: string | null;
  productCount: number;
  childrenCount: number;
  createdAt: string;
  updatedAt: string | null;
}

export interface CategoryCreateDto {
  name: string;
  description?: string;
  slug?: string;
  parentId?: number;
  isActive?: boolean;
  displayOrder?: number;
  imageUrl?: string;
  metaTitle?: string;
  metaDescription?: string;
}
